### ISSUE TITLE
instance-debootstrap needs to refuse too-small disk

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#220]
Originally reported of Google Code with ID 164.

```
on squeeze, with debootstrap from your git repository

root@testserver1:/usr/local/lib/ganeti/tools# gnt-cluster --version
gnt-cluster (ganeti v2.4.2) 2.4.2

root@testserver1:/usr/local/lib/ganeti/tools# gnt-cluster version
Software version: 2.4.2
Internode protocol: 2040000
Configuration format: 2040000
OS api version: 20
Export interface: 0


root@testserver1:~# cd /usr/local/lib/ganeti/tools/
root@testserver1:/usr/local/lib/ganeti/tools# ./burnin -o debootstrap+default ganeti{1..3}                                 - Testing global parameters
- Creating instances
  * instance ganeti1
    on testserver1.AAA.localdomain, testserver2.AAA.localdomain
- Non-idempotent opcode failed, aborting
- Error detected: opcode buffer follows:


- Sun Jun  5 23:46:45 2011 * creating instance disks...
Sun Jun  5 23:46:48 2011 adding instance ganeti1.AAA.localdomain to cluster config
Sun Jun  5 23:46:48 2011  - INFO: Waiting for instance ganeti1.AAA.localdomain to sync disks.
Sun Jun  5 23:46:49 2011  - INFO: - device disk/0:  9.10% done, 28s remaining (estimated)
Sun Jun  5 23:47:17 2011  - INFO: Instance ganeti1.AAA.localdomain's disks are in sync.
Sun Jun  5 23:47:17 2011 * running the instance OS create scripts...

- 


- Removing instances
Traceback (most recent call last):
  File "./burnin", line 1081, in <module>
    main()
  File "./burnin", line 1077, in main
    return Burner().BurninCluster()
  File "./burnin", line 989, in BurninCluster
    self.BurnCreateInstances()
  File "./burnin", line 228, in wrapper
    val = fn(self, *args, **kwargs)
  File "./burnin", line 248, in batched
    val = fn(self, *args, **kwargs)
  File "./burnin", line 583, in BurnCreateInstances
    self.ExecOrQueue(instance, [op], post_process=remove_instance(instance))
  File "./burnin", line 356, in ExecOrQueue
    val = self.ExecOp(self.queue_retry, *ops) # pylint: disable-msg=W0142
  File "./burnin", line 348, in ExecOp
    return self.MaybeRetry(rval, "opcode", self._ExecOp, *ops)
  File "./burnin", line 305, in MaybeRetry
    val = fn(*args)
  File "./burnin", line 330, in _ExecOp
    results = cli.PollJob(job_id, cl=self.cl, feedback_fn=self.Feedback)
  File "/usr/local/lib/python2.6/dist-packages/ganeti/cli.py", line 1708, in PollJob
    return GenericPollJob(job_id, _LuxiJobPollCb(cl), reporter)
  File "/usr/local/lib/python2.6/dist-packages/ganeti/cli.py", line 1530, in GenericPollJob
    errors.MaybeRaise(msg)
  File "/usr/local/lib/python2.6/dist-packages/ganeti/errors.py", line 448, in MaybeRaise
    raise errcls, args
ganeti.errors.OpExecError: Could not add os for instance ganeti1.AAA.localdomain on node testserver1.AAA.localdomain: OS create script failed (exited with exit code 2), last lines in the log file:
tar: ./dev/ram4: Cannot mknod: No such file or directory
tar: ./dev/mpu401data: Cannot mknod: No such file or directory
tar: ./dev/port: Cannot mknod: No such file or directory
tar: ./dev/full: Cannot mknod: No such file or directory
tar: ./dev/rmidi2: Cannot mknod: No such file or directory
tar: ./dev/tty5: Cannot mknod: No such file or directory
tar: ./dev/midi2: Cannot mknod: No such file or directory
tar: ./dev/agpgart: Cannot mknod: No such file or directory
tar: ./dev/mpu401stat: Cannot mknod: No such file or directory
tar: ./dev/loop5: Cannot mknod: No such file or directory
tar: ./dev/tty8: Cannot mknod: No such file or directory
tar: ./dev/loop3: Cannot mknod: No such file or directory
tar: ./dev/ram2: Cannot mknod: No such file or directory
tar: ./dev/loop6: Cannot mknod: No such file or directory
tar: ./dev/dsp1: Cannot mknod: No such file or directory
tar: ./dev/midi02: Cannot mknod: No such file or directory
tar: ./dev/fd: Cannot open: No such file or directory
tar: ./dev/pts: Cannot mkdir: No such file or directory
tar: ./dev/mem: Cannot mknod: No such file or directory
tar: Exiting with failure status due to previous errors

```
Originally added on 2011-06-05 21:51:30 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
p.s. I didn't installed ganeti from packages, but from the online tar.gz ... I noticed afterwards that Debian had packages available (I blindly followed the online "german" and full version installation/administration guide)
```
Originally added on 2011-06-05 21:58:57 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Have you tried with a bigger disk size? It could be that the default is simply too small.
```
Originally added on 2011-06-06 09:04:35 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
indeed... it was that :)
```
Originally added on 2011-06-06 09:25:37 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
it was that, u can close the ticket :)
```
Originally added on 2011-06-06 09:25:52 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Well, I think that instance-debootstrap needs to check the disk size and refuse to start with < 256MB or similar. So I'll change the title of this issue accordingly, thanks!
```
Originally added on 2011-06-06 09:28:39 +0000 UTC.

Changed State: Accepted

**Comment by Ganeti-Issues-Migrator** [comment]
```
p.s. consider that the ./burnin is now running, it is creating the 
second box, but the first one has booted but locked itself after 
starting the crond... If I get the second instance I copy u here the 
console output
```
Originally added on 2011-06-06 09:31:28 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
the console from one of the machines... I do not get to a login screen...

    [    1.021874] md: raid5 personality registered for level 5
    [    1.021879] md: raid4 personality registered for level 4
    Success: loaded module raid456.
    Success: loaded module raid5.
    Success: loaded module raid6.
    [    1.030420] md: raid10 personality registered for level 10
    Success: loaded module raid10.
    done.
    Begin: Assembling all MD arrays ... mdadm: No arrays found in config
    file or automatically
    Failure: failed to assemble all arrays.
    done.
    [    1.047003] device-mapper: uevent: version 1.0.3
    [    1.047501] device-mapper: ioctl: 4.15.0-ioctl (2009-04-01)
    initialised: dm-devel@redhat.com
    done.
    Begin: Running /scripts/local-premount ... done.
    [    1.076390] kjournald starting.  Commit interval 5 seconds
    [    1.076405] EXT3-fs: mounted filesystem with ordered data mode.
    Begin: Running /scripts/local-bottom ... done.
    done.
    Begin: Running /scripts/init-bottom ... done.
    modprobe: FATAL: Could not load
    /lib/modules/2.6.32-5-xen-amd64/modules.dep: No such file or directory

    INIT: version 2.86 booting
    Mount point '/dev/shm' does not exist. Skipping mount. (warning).
    Setting the system clock.
    Cannot access the Hardware Clock via any known method.
    Use the --debug option to see the details of our search for an
    access method.
    Unable to set System Clock to: Mon Jun 6 09:35:43 UTC 2011 (warning).
    Activating swap...done.
    Checking root file system...fsck 1.41.3 (12-Oct-2008)
    /dev/xvda1: clean, 8624/65280 files, 45378/261056 blocks
    done.
    [    2.094280] EXT3 FS on xvda1, internal journal
    Setting the system clock.
    Cannot access the Hardware Clock via any known method.
    Use the --debug option to see the details of our search for an
    access method.
    Unable to set System Clock to: Mon Jun 6 09:35:43 UTC 2011 (warning).
    Cleaning up ifupdown....
    Loading kernel modules...FATAL: Could not load
    /lib/modules/2.6.32-5-xen-amd64/modules.dep: No such file or directory
    Checking file systems...fsck 1.41.3 (12-Oct-2008)
    done.
    Setting kernel variables (/etc/sysctl.conf)...done.
    Mounting local filesystems...done.
    Activating swapfile swap...done.
    Setting up networking....
    Configuring network interfaces...done.
    INIT: Entering runlevel: 2
    Starting enhanced syslogd: rsyslogd.
    Starting periodic command scheduler: crond.
```
Originally added on 2011-06-06 09:40:59 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2012-07-19 13:38:11 +0000 UTC.

Added Labels: Type-Defect Component-instances

**Comment by Ganeti-Issues-Migrator** [comment]
```
For the last one, the problem is with the console definition, issue 284.
```
Originally added on 2012-10-30 10:12:32 +0000 UTC.

Mentioned Issue with GoogleCode ID 284 has been migrated to GitHub with ID #340.
Changed State: Started

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2012-10-30 10:16:53 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released in version 0.12.
```
Originally added on 2012-10-30 12:08:45 +0000 UTC.

Changed State: Released
